src/components/atoms/BaseCard/BaseCard.tsx

```diff
@@ -67,17 +96,68 @@ export const BaseCard = forwardRef<CardHandle, BaseCardProps>(
       flip: (forceFaceUp?: boolean) => turnCard(forceFaceUp),
     }));
 
-    const onClick = () => !disableNativeEvents && turnCard();
+    const onClick = () => !disableNativeEvents && !isDragging && Date.now() - startDate < 200 && turnCard();
+
     const onKeyDown = (event: KeyboardEvent<HTMLDivElement>) =>
       !disableNativeEvents && event.keyCode === 32 && turnCard();
 
+    const handleMouseDown = useCallback(
+      ({ clientX, clientY }) => {
+        if (!disableNativeEvents) {
+          setStartDate(Date.now());
+          setStartPosition({ x: clientX, y: clientY });
+          setIsDraging(true);
+        }
+      },
+      [disableNativeEvents]
+    );
+
+    const handleMouseMove = useCallback(
+      ({ clientX, clientY }) => {
+        const newCardPosition: Position = {
+          x: Math.round(cardPosition.x + clientX - startPosition.x),
+          y: Math.round(cardPosition.y + clientY - startPosition.y),
+        };
+        setCardPosition(newCardPosition);
+      },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [startPosition]
+    );
+
+    const handleMouseUp = useCallback(() => {
+      setIsDraging(false);
+    }, []);
+
+    useEffect(() => {
+      if (isDragging) {
+        window.addEventListener('mousemove', handleMouseMove);
+        window.addEventListener('mouseup', handleMouseUp);
+      } else {
+        window.removeEventListener('mousemove', handleMouseMove);
+        window.removeEventListener('mouseup', handleMouseUp);
+
+        onDragEnd(cardPosition);
+      }
+    }, [isDragging, handleMouseMove, handleMouseUp, cardPosition, onDragEnd]);
+
     return (
-      <div data-testid="BaseCard" role="button" onClick={onClick} onKeyDown={onKeyDown} className={card} tabIndex={0}>
-        <div data-testid="BaseCard-flipper" className={classnames(cardFlipper, { [flipped]: !isFrontVisible })}>
-          <div data-testid="BaseCard-front" className={classnames(cardFront)}>
+      <div
+        data-testid="BaseCard"
+        role="button"
+        onClick={onClick}
+        onKeyDown={onKeyDown}
```

Comment:
Question: Do we want to have the onKeyDown activated if I am dragging the card?

Reply:
I would say yes, why not? I don't find it disturbing